### ISSUE TITLE
Components relying on `locale` state aren't fully tested to account for `undefined` value

### ISSUE THREAD

**Issue by joeyparis** [react-boilerplate/react-boilerplate#2824]
Before opening a new issue, please take a moment to review our [**community guidelines**](https://github.com/react-boilerplate/react-boilerplate/blob/master/CONTRIBUTING.md) to make the contribution process easy and effective for everyone involved.

## Description

When working on #2832 I realized that I was deconstructing the state selector for `selectLocale` when I shouldn't have been so even though `locale` was defined as `en`, `undefined` was being passed to `react-intl`. Despite this error, all tests passed successfully. It was "logged" while running the tests since `react-intl` will fall back to the default locale if it doesn't find the passed parameter, but doesn't show up in the tests report.

## Steps to reproduce

Attempt to deconstruct `useSelector(selectLocale)` in any component that uses it, and run the tests. You will see the fallback error message from `react-intl` but the tests will still be marked as passed. Here's [the commit](https://github.com/react-boilerplate/react-boilerplate/pull/2823/commits/26617ba532747b492c2fb1b8d50878420acad80f) where I fixed this for a direct example.
![Commit changes](https://imgur.com/co4OfKK.png)

**Expected behavior**

The test should fail if the `locale` state is defined, but is not passed to the relative children component. Of course, if the `locale` state is undefined then the error produced by `react-intl` is accurate and shouldn't result in a failed test. 

**Screenshots**

![locale error pass](https://imgur.com/dmikFL4.png)

## Versions

- React-Boilerplate: 4.0.0
- Node/NPM: 10.16.0
- Browser: N/A

**Comment by julienben** [comment]
I don't think this is a big issue but if you'd like to add a test for this in #2823, you're more than welcome to. (The LanguageProvider tests are reused between the template and sample apps so you only to add it once.)
